executor/replace.go

```diff
@@ -89,6 +91,33 @@ func (e *ReplaceExec) removeRow(ctx context.Context, txn kv.Transaction, handle
 	return false, nil
 }
 
+// EqualDatums compare if a and b contains the same datum values.
+func (e *ReplaceExec) EqualDatums(sc *stmtctx.StatementContext, a []types.Datum, b []types.Datum) (bool, error) {
+	if len(a) != len(b) {
+		return false, nil
+	}
+	if a == nil && b == nil {
```

Comment:
We better not check nil for slice type because it's very common we mix the use of nil slice and not-nil zero len slice.